### ISSUE TITLE
1st argument is not a valid UUID

### ISSUE THREAD

**Issue by maxbrito** [das-labor/panopticon#221]
Hello,

I'm unable to open any executables when running on OSX (10.11.4) using the compiled binary for version 0.16, this is the error message:
![Image of screen1](http://image.prntscr.com/image/d8a1bca6971e42e7beab29325571a8f2.png)

This happened for every Windows executable that I've tried opening. Also tried with a few Linux executables and similar error box.

**Comment by hashedone** [comment]
Exactly same happens to me on Windows 10, with windows binaries. Panopticon version 0.16.0, binary i tried to disassemble is compilled by rust compiler.

**Comment by lindt** [comment]
If your local installation has problems, you could try the docker image. Should work on mac and on win. I just tested it on linux.

```
docker run -ti -e DISPLAY \
  --volume "$HOME/.Xauthority:/root/.Xauthority" \
  --volume "/tmp/.X11-unix:/tmp/.X11-unix:rw" \
  --volume "$(pwd):/work"\
  --workdir /work \
  --net=host \
  think/panopticon
```

**Comment by m4b** [comment]
So afaik panopticon 0.16 didn't ship with a functioning PE backend, which is why you get the above error (and the error message could be more useful, i.e., something like unsupported format in a dialog box (PR anyone?))

I've been promising @flanfly a PE backend in goblin for several months so it's time I stopped lying about ;)

I think I can get a functioning backend for windows PE binaries done this weekend, but it will require finalizing some other crates I'm working on which goblin will use (and which I keep obsessing over but it's time to stop).

This means that panopticon master should have a PE backend (which means you can run it on windows binaries), but for the binary releases we'll probably need to ship it with 0.17.  However, I don't what @flanfly 's plans for next release are?

Anyway, stay tuned :P

**Comment by nunobrito** [comment]
I think this is the same in Linux. Tried with a recent Linux Mint distro, what should we have installed on a apt-get based distro? Thanks.

**Comment by m4b** [comment]
@nunobrito So opening up a PE file (the name for a windows binary format) on a Linux machine _will_ be possible (similarly opening up a PE file on a windows machine) once the PE loader is done.

However, if you're saying you get the above error opening up an ELF binary (the binary format used on Linux), then that definitely is an error, and more info would be appreciated.

**Comment by flanfly** [comment]
Hey @maxbrito. For now, Panopticon is only able to diassemble ELF binaries. So OSX and Windows applications won't work.

You can see whenever Panopticon recognizes the file in the file picker. If it says "ELF" on the right end of the line, the file should work. The "... not a valid UUID" error sometimes shows up even when disassembly has worked (I suspect a race-condition somewhere between the Panopticon core and the front-end that runs in a different thread :sob:). Just click on one of the functions listed on the left side. If there is none, you likely opened a file Panopticon does not know what to do with.

You can also check whenever something is an ELF file by using the `file` command. `file <path to binary>` should print something like `ELF 64-bit LSB executable`.

**Comment by maxbrito** [comment]
Thanks for the feedback. I thought to have made a mistake but then verified again and still got the same output as before when trying to open a Linux executable.

![screenshot](http://image.prntscr.com/image/ee0ba404bb1c4bcf8edabb49b4db646e.png)

**Comment by m4b** [comment]
Uh, that doesn't seem good. How large is the binary (`du -h masterpdfeditor3`)? I've noticed some slowdowns on my machine when the binary is pretty huge

**Comment by maxbrito** [comment]
19.3Mb

Hadn't noticed myself it was that big. Case solved.

**Comment by flanfly** [comment]
The performance issue should be resolved w/ #283. Please reopen if it's still slow.
